lib/krikri/harvesters/oai_harvester.rb

```diff
@@ -207,5 +218,31 @@ def record_xml(rec)
         .add_namespace_definition(nil, 'http://www.openarchives.org/OAI/2.0/')
       doc.to_xml
     end
+
+    private
+
+    def get_identifier(record)
+      return record.header.identifier unless @id_path
+
+      xml = Nokogiri.XML(record_xml(record))
```

Comment:
I notice that we have to call `Nokogiri.XML` here because the return value of `record_xml` is a string, via its `doc.to_xml`.  But `Nokokiri.XML` is already called in `record_xml` in https://github.com/dpla/KriKri/blob/feature/oai-id-selection/lib/krikri/harvesters/oai_harvester.rb#L216 ... Would it be worth it to refactor so that the unserialization doesn't have to happen twice?